File: typescript/services/InvestorNotifier.ts
import { BitcoinPriceObserver } from "./BitcoinPriceObserver";

export class InvestorNotifier implements BitcoinPriceObserver {
    update(oldPrice: number, newPrice: number) {
        console.log('\n\nInvestorNotifier');
        console.log(`Price updated: ${oldPrice} -> ${newPrice}`);
    }
}

File: src/main/java/com/douglasmatosdev/service/InvestorNotifier.java
package com.douglasmatosdev.service;

public class InvestorNotifier implements BitcoinPriceObserver {

    @Override
    public void update(Double oldPrice, Double newPrice) {
        if (newPrice > oldPrice) {
            System.out.println("Price increased! Notify investors!");
        } else {
            System.out.println("Price decreased! Notify investors!");
        }
    }
}
